src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -140,20 +142,246 @@ points distance from the plane is negative) using @ref Distance::pointPlaneScale
+points distance larger than the sphere's radius) using @ref Distance::pointPlaneScaled().
+*/
+template<class T> bool sphereFrustum(const Vector3<T>& center, T radius, const Frustum<T>& frustum);
+
+/**
+@brief Intersection of a point and a cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param angle    Apex angle of the cone
+
+Returns @cpp true @ce if the point is inside the cone.
+
+@see @ref pointCone(const Vector3&, const Vector3&, const Vector3&, T)
+*/
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, Rad<T> angle);
+
+/**
+@brief Intersection of a point and a cone using precomputed values
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation:
+                         @cpp T(1) + Math::pow(Math::tan(angle/T(2)), T(2)) @ce
+
+Returns @cpp true @ce if the point is inside the cone.
+*/
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, T tanAngleSqPlusOne);
+
+/**
+@brief Intersection of a point and a double cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param angle    Apex angle of the cone
+
+Returns @cpp true @ce if the point is inside the double cone.
+
+@see @ref pointDoubleCone(const Vector3&, const Vector3&, const Vector3&, T)
+*/
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, Rad<T> angle);
+
+/**
+@brief Intersection of a point and a double cone using precomputed values
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation:
+                         @cpp T(1) + Math::pow(Math::tan(angle/T(2)), T(2)) @ce
+
+Returns @cpp true @ce if the point is inside the double cone.
+
+Uses the result of precomputing @f$ x = \tan^2{\theta} + 1 @f$.
+*/
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, T tanAngleSqPlusOne);
+
+/**
+@brief Intersection of a sphere and a cone view
+@param sphereCenter Center of the sphere
+@param radius Radius of the sphere
+@param coneView View matrix with translation and rotation of the cone
+@param angle Cone opening angle
+
+Returns @cpp true @ce if the sphere intersects the cone.
+
+Transforms the sphere center into cone space (using the cone view matrix) and
+performs sphere-cone intersection with the zero-origin -Z axis-aligned cone.
+
+@see @ref sphereConeView(const Vector3<T>&, T, const Matrix4<T>&, T, T, T)
+*/
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, T radius, const Matrix4<T>& coneView, Rad<T> angle);
```

Comment:
Does this work as expected even with the pathological matrices correctly? Such as a skew matrix or non-uniform scaling?

If not, I'd maybe use a some more restricted representation ... or at least mention it in the docs.

Reply:
It will not, it expect it to be a rigid transform with no or uniform scaling.
Restricted representation as in rotation matrix + translation? Or Dual quaternion? Or Assert on [isRigidTransformation()](http://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1Matrix3.html#a267cce4346f18620cd93df07bf858234)?

Reply:
Hmm. What would be the best in terms of usability? 

- Allowing Matrix4 with no check inside would lead to surprising results very often, considering scaling is usually part of a transformation. 3x3 rotation matrix suffers from the same problem.
- Assert on `isRigidTransformation()` might be overly expensive
- Requiring dual quaternion might feel too alien to some people

Hmm...

Reply:
Considering this is used (if at all) with either camera transformation matrices or light transformation matrices, I would expect the isRigidTransformation to be the most sensible as all other would require a conversion.